services/src/unit-test/java/uk/gov/moj/sdt/services/SubmitQueryServiceTest.java

```diff
@@ -259,121 +234,101 @@ public void testSubmitQueryRequestSoapFault() {
-        receiveTimeOutParam.setName("TARGET_APP_RESP_TIMEOUT");
+        receiveTimeOutParam.setName(TARGET_APP_RESP_TIMEOUT);
         receiveTimeOutParam.setValue("12000");
-        EasyMock.expect(this.mockGlobalParamCache.getValue(IGlobalParameter.class, "TARGET_APP_RESP_TIMEOUT"))
-                .andReturn(receiveTimeOutParam);
+        when(this.mockGlobalParamCache.getValue(IGlobalParameter.class, TARGET_APP_RESP_TIMEOUT))
+                .thenReturn(receiveTimeOutParam);
 
         final IGlobalParameter maxQueryReq = new GlobalParameter();
         maxQueryReq.setName("MCOL_MAX_CONCURRENT_QUERY_REQ");
         maxQueryReq.setValue("5");
-        EasyMock.expect(this.mockGlobalParamCache.getValue(IGlobalParameter.class, "MCOL_MAX_CONCURRENT_QUERY_REQ"))
-                .andReturn(maxQueryReq);
+        when(this.mockGlobalParamCache.getValue(IGlobalParameter.class, "MCOL_MAX_CONCURRENT_QUERY_REQ"))
+                .thenReturn(maxQueryReq);
 
         final WebServiceException wsException = new WebServiceException("WS_ERROR");
-        this.mockConsumerGateway.submitQuery(submitQueryRequest, 1000, 12000);
-        EasyMock.expectLastCall().andThrow(wsException);
+
+        doThrow(wsException).when(mockConsumerGateway).submitQuery(submitQueryRequest, 1000, 12000);
 
         final IErrorMessage errorMsg = new ErrorMessage();
         errorMsg.setErrorCode("SDT_INT_ERR");
         errorMsg.setErrorDescription("SDT Internal Error.");
         errorMsg.setErrorText("SDT Internal Error, please report to {0}");
-        EasyMock.expect(this.mockErrorMsgCacheable.getValue(IErrorMessage.class, "SDT_INT_ERR")).andReturn(errorMsg);
+        when(this.mockErrorMsgCacheable.getValue(IErrorMessage.class, "SDT_INT_ERR")).thenReturn(errorMsg);
 
         final IGlobalParameter contactNameParameter = new GlobalParameter();
         contactNameParameter.setValue("Tester");
         contactNameParameter.setName("CONTACT_DETAILS");
-        EasyMock.expect(this.mockGlobalParamCache.getValue(IGlobalParameter.class, "CONTACT_DETAILS")).andReturn(
+        when(this.mockGlobalParamCache.getValue(IGlobalParameter.class, "CONTACT_DETAILS")).thenReturn(
                 contactNameParameter);
 
-        EasyMock.replay(mockConsumerGateway);
-        EasyMock.replay(mockGlobalParamCache);
-        EasyMock.replay(mockErrorMsgCacheable);
-        EasyMock.replay(mockBulkCustomerDao);
-
         SdtContext.getContext().setRawInXml("criteria");
         this.submitQueryService.submitQuery(submitQueryRequest);
 
-        EasyMock.verify(mockConsumerGateway);
-        EasyMock.verify(mockGlobalParamCache);
-        EasyMock.verify(mockErrorMsgCacheable);
-        EasyMock.verify(mockBulkCustomerDao);
-
-        Assert.assertEquals("Raw output xml should be null", null, SdtContext.getContext().getRawOutXml());
+        verify(mockConsumerGateway).submitQuery(submitQueryRequest, 1000, 12000);
```

Comment:
Need to verify behaviour of other mock objects (mockGlobalParamCache and mockErrorMsgCacheable).

Reply:
done